gen/gen.go

```diff
@@ -276,7 +276,8 @@ func (g *Generator) resolveRecordSchema(schema *avro.RecordSchema) string {
 
 	typeName := g.resolveTypeName(schema)
 	if !g.hasTypeDef(typeName) {
-		g.typedefs = append(g.typedefs, newType(typeName, schema.Doc(), fields, schema.String()))
+		schemaBytes, _ := schema.MarshalJSON()
+		g.typedefs = append(g.typedefs, newType(typeName, schema.Doc(), fields, string(schemaBytes)))
```

Comment:
Funny that the stringify does not include default value instead of JSON serialization.

Reply:
Yeah, the string function returns the canonical form of the schema which does not include the default value sadly.
See: https://github.com/hamba/avro/issues/115 for reference which is the issue that gave birth to the MarshalJSON function